rod_cutting_problem/rod_cutting.c

```diff
@@ -0,0 +1,31 @@
+/*
+*  Given a rod of length n length unit and an array of prices that contains prices of all pieces of size smaller than n. 
+*  Determine the maximum new_pricesue obtainable by cutting up the rod and selling the pieces.
+*  Time Complexity: O(n^2)
+*  Space Complexity: O(n)
+*/
+
+#include <stdio.h>
+#include <string.h>
+
+#define max(a, b) (a > b)? a : b;
+
+int rod_cutting(int *prices) {
+    int i, j;
+    int n = sizeof(prices);
+    int new_prices[n + 1];
+    memset(new_prices, 0, n + 1);
+    for (i = 1; i < n + 1; ++i) {
+        int maxi = -1;
+        for (j = 0; j < i; ++j)
+            maxi = max(maxi, prices[j] + new_prices[i - j - 1]);
```

Comment:
Pass the second argument in parantheses as you haven't defined that in the macro. It will be treated as
```c
(maxi > prices[j] + new_prices[i - j - 1])? maxi : prices[j] + new_prices[i - j - 1];
```
But if you put a parantheses, it will become
```c
(maxi > (prices[j] + new_prices[i - j - 1]))? maxi : (prices[j] + new_prices[i - j - 1]);
```